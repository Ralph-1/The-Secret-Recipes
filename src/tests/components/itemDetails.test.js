import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import RecipeDescription from '../../components/Description';

const initial = {
  category: 'test',
  area: 'jest',
  ingredients: ['react', 'redux'],
  className: 'component',
};

const {
  ingredients, category, area, className,
} = initial;

Enzyme.configure({ adapter: new Adapter() });
const setup = () => {
  const component = shallow(
    <RecipeDescription
      ingredients={ingredients}
      category={category}
      area={area}
      className={className}
    />,
  );
  return component;
};

describe('Header Component', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<RecipeDescription
      ingredients={ingredients}
      category={category}
      area={area}
      className={className}
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  let component;
  beforeEach(() => {
    component = setup();
  });

  it('should render one div', () => {
    expect(component.find('div').length).toBe(2);
  });

  it('should always render a div', () => {
    expect(component.find('div').length).not.toBe(0);
  });

  it('should render 6 span element', () => {
    expect(component.find('span').length).toBe(6);
  });

  it('should always render a span', () => {
    expect(component.find('span').length).not.toBe(0);
  });
});
