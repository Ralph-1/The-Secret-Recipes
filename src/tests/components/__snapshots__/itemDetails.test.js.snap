// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match snapshot 1`] = `
<div
  className="component"
>
  <p
    className="pt-4"
  >
    <span
      className="font-weight-bolder h5"
    >
      Category :
    </span>
    <span>
      test
    </span>
  </p>
  <br />
  <hr />
  <p
    className="pt-4"
  >
    <span
      className="font-weight-bolder h5"
    >
      Area :
    </span>
    <span>
      jest
    </span>
  </p>
  <br />
  <hr />
  <p
    className="pt-4 font-weight-bolder h5"
  >
    Ingredients :
  </p>
  <div
    className="pb-4"
  >
    <span>
      react
       
      |
       
    </span>
    <span>
      redux
       
      |
       
    </span>
  </div>
  <hr />
</div>
`;
